tests/utils/fixtures/fixtures.py

```diff
@@ -155,14 +166,20 @@ def qemu_finalizer_impl(conn):
                 conn.run("poweroff")
                 halt_time = time.time()
                 # Wait up to 30 seconds for shutdown.
-                while halt_time + 30 > time.time() and qemu.poll() is None:
+                while (
+                    halt_time + shutdown_timeout_s > time.time() and qemu.poll() is None
+                ):
                     time.sleep(1)
             except:
                 pass
 
             # Terminate qemu
             try:
-                qemu.terminate()
+                os.killpg(os.getpgid(qemu.pid), signal.SIGTERM)
+                while pid_exists(qemu.pid) and shutdown_timeout_s > 0:
+                    os.killpg(os.getpgid(qemu.pid), signal.SIGKILL)
+                    time.sleep(1)
```

Comment:
If you want to try `SIGKILL` after waiting for one second, you need to put the sleep before the kill. Otherwise this code will race from the `SIGTERM` to the `SIGKILL` in no time, almost certainly before qemu has had time to exit.